crates/hir/src/term_search/expr.rs

```diff
@@ -209,7 +209,7 @@ impl Expr {
                 }
             }
             Expr::Variant { variant, generics, params } => {
-                let generics = non_default_generics(db, (*variant).into(), generics);
+                let generics = non_default_generics(db, variant.parent_enum(db).into(), generics);
```

Comment:
@kilpkonn heads up but the logic here is/was wrong. `non_default_generics` only queries the generics of the definition, not the parent so querying it for an enum variant always resulted in an empty set of generics. Hence why this now changes the test outputs for generics in the assist below. (technically this is not more correct)